File: Level 2/행렬의 곱셈/행렬의 곱셈.py
def productMatrix(A, B):
    answer = []
    tmp_Sum = 0
    for i in range(len(A)):
        tmp = []
        for j in range(len(A[0])):
            tmp_Sum = 0
            for k in range(len(B)):
                tmp_Sum += A[i][k] * B[k][j]
            tmp.append(tmp_Sum)
        answer.append(tmp)
    return answer

# 아래는 테스트로 출력해 보기 위한 코드입니다.
a = [ [ 1, 2 ], [ 2, 3 ]]
b = [[ 3, 4], [5, 6]]
print("결과 : {}".format(productMatrix(a,b)))

File: Level 2/행렬의 곱셈/행렬의 곱셈.cpp
#include<iostream>
#include<vector>
using namespace std;

vector<vector<int> >productMatrix(vector<vector<int> >A, vector<vector<int> >B)
{
	vector<vector<int> >answer;
  vector<int> tmp;
    int tmp_Sum = 0;
    int i, j, k;
 		tmp.clear();
    for (i = 0; i < A.size(); i++){
       tmp.clear();
        for (j = 0; j < B[0].size(); j++){
            tmp_Sum = 0;
            for (k = 0; k < A[0].size(); k++){
            	tmp_Sum += A[i][k] * B[k][j];
            }
            tmp.push_back(tmp_Sum);
        }
        answer.push_back(tmp);
    }
    return answer;
}

int main()
{
	vector<vector<int> >A{{1,2},{2,3}};
	vector<vector<int> >B{{2,3},{3,4}};
	vector<vector<int> > testAnswer = productMatrix(A,B);

	for(int i=0;i<testAnswer.size(); i++)
	{
		for(int j=0;j<testAnswer[i].size(); j++)
			cout<<testAnswer[i][j]<<" ";
		cout<<"\n";
	}
}
